### PR TITLE
Implement Spring Security

### PR DESCRIPTION
**Pull request by toshydev**
- add SecurityConfig
- add securityFilterChain
- add MongoUser
- Argon2PasswordEncoder incl. Dependency
- add MongoUserController (principal)
- add csrf and Integrationtests with mockuser
- add SecurityConfig to ConfigTest annotation and add root path requestMatcher to Security Config
- add tests for security package and integration test
- add Login form and button and display the logged-in user in the Header
- add Logout button

### CODE REVIEW

File: backend/src/main/java/org/partypets/backend/security/MongoUserController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        return "AnonymousUser";
+    }
+
+    @GetMapping("/me2")
+    public String getMe2() {
+        return SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+    }
```
Comment (Ben-21) [inline backend/src/main/java/org/partypets/backend/security/MongoUserController.java:33]: Both endpoints are doing the same. You can consider deleting one of them.

File: backend/src/test/java/org/partypets/backend/security/MongoUserDetailServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        MongoUser expected = new MongoUser("abc", "Henry", "p4ssw0rd");
+        String username = "Henry";
+        //WHEN
+        when(userRepository.findByUsername(username)).thenReturn(Optional.of(expected));
+        UserDetails actual = userDetailService.loadUserByUsername("Henry");
+        //THEN
+        assertEquals(expected.username(), actual.getUsername());
+        verify(userRepository).findByUsername(username);
+    }
+}
```
Comment (Ben-21) [inline backend/src/test/java/org/partypets/backend/security/MongoUserDetailServiceTest.java:31]: If you add an empty line at the end of your code, you don't get this awkward little red symbol ;-)

**Review by Ben-21 — APPROVED** [review]
LSGTU

File: backend/src/main/java/org/partypets/backend/security/MongoUserController.java (hunk 1 above)
Comment (toshydev) [reply backend/src/main/java/org/partypets/backend/security/MongoUserController.java:33]: True, it was meant for learning purpose. We'll delete one of them👌

File: backend/src/test/java/org/partypets/backend/security/MongoUserDetailServiceTest.java (hunk 2 above)
Comment (toshydev) [reply backend/src/test/java/org/partypets/backend/security/MongoUserDetailServiceTest.java:31]: We keep forgetting those 😅

**Pull request merged**